source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+				&(txBuffDescrip_0[0]); /* Aligned transmit buffer descriptor start address. */
+		buffCfg[0].rxBufferAlign =
+				&(rxDataBuff_0[0][0]); /* Receive data buffer start address. */
+		buffCfg[0].txBufferAlign = &(txDataBuff_0[0][0]); /* Transmit data buffer start address. */
+		buffCfg[0].txFrameInfo = NULL; /* Transmit frame information start address. Set only if using zero-copy transmit. */
+		buffCfg[0].rxMaintainEnable = true; /* Receive buffer cache maintain. */
+		buffCfg[0].txMaintainEnable = true; /* Transmit buffer cache maintain. */
+
+		sysClock = phyHandle.mdioHandle->resource.csrClock_Hz;
+
+		ENET_GetDefaultConfig(&config);
+
+		config.ringNum     = ENET_RING_NUM;
+		config.rxBuffAlloc = NULL;
+		config.rxBuffFree  = NULL;
+		config.userData    = ethernetifLocal;
+        config.miiSpeed    = (enet_mii_speed_t)speed;
+	    config.miiDuplex   = (enet_mii_duplex_t)duplex;
+
+		#if USE_RTOS && defined(SDK_OS_FREE_RTOS)
+		    uint32_t instance;
+		    static ENET_Type *const enetBases[]  = ENET_BASE_PTRS;
+		    static const IRQn_Type enetTxIrqId[] = ENET_Transmit_IRQS;
+		    /*! @brief Pointers to enet receive IRQ number for each instance. */
+		    static const IRQn_Type enetRxIrqId[] = ENET_Receive_IRQS;
+
+		    /* Only get interrupt for incoming messages. */
+		    config.interrupt = kENET_RxFrameInterrupt;
+		    config.callback = ethernet_callback;
+
+		    for (instance = 0; instance < ARRAY_SIZE(enetBases); instance++)
+		    {
+		        if (enetBases[instance] == ethernetifLocal->base)
+		        {
+		            NVIC_SetPriority(enetRxIrqId[instance], ENET_PRIORITY);
+		            NVIC_SetPriority(enetTxIrqId[instance], ENET_PRIORITY);
+		            break;
+		        }
+		    }
+
+		    configASSERT( instance != ARRAY_SIZE( enetBases ) );
+		#endif /* USE_RTOS */
+
+		    for (i = 0; i < ENET_RXBUFF_NUM; i++)
+		    {
+		    	ethernetifLocal->RxPbufs[i].buffer                 = &(ethernetifLocal->RxDataBuff[i][0]);
+		    	ethernetifLocal->RxPbufs[i].buffer_used            = false;
+		    }
+
+		    /* Initialize the ENET module. */
+		    /* TODO: set MAC Address properly. */
+		    ENET_Init( ethernetifLocal->base, &ethernetifLocal->handle, &config, &buffCfg[0], ipLOCAL_MAC_ADDRESS, sysClock );
+
+		    ENET_ActiveRead( ethernetifLocal->base );
+	}
+
+	if( xResult != pdFAIL )
+	{
+		/* Enable the interrupt and set its priority to the minimum
+		 * interrupt priority.  */
+		NVIC_SetPriority( ENET_IRQn, ENET_PRIORITY );
+		NVIC_EnableIRQ( ENET_IRQn );
+	}
+
+	return xResult;
+}
+
+BaseType_t xNetworkInterfaceOutput( NetworkBufferDescriptor_t * const pxNetworkBuffer,
+                                    BaseType_t xReleaseAfterSend )
+{
+	status_t result;
+	BaseType_t xReturn = pdFAIL;
+	bool LinkUp = false;
+	status_t readStatus;
+
+	readStatus = PHY_GetLinkStatus( &phyHandle, &LinkUp );
+
+	if( ( readStatus == kStatus_Success ) &&
+		( LinkUp == true ) )
+	{
+		result = ENET_SendFrame( ethernetifLocal->base,
```

Comment:
Does `ENET_SendFrame()` wait until the frame has been sent?
Or if not, does it send a copy of the frame?

Reply:
It copies the frame